docs/upgrade/4.x.x.adoc

```diff
@@ -0,0 +1,475 @@
+= React4XP - Upgrade - 4.x.x
+:toc: right
+
+== What
+
+=== Graal.JS
+
+We have dropped support for Nashorn in favour of Graal.JS.
+
+This means we had to upgrade the Enonic XP version requirement to 7.12
```

Comment:
Since XP >= 7.12 is required, this should also be mentioned on how.

Reply:
Added gradle.properties section